kafka-client-lib/src/main/java/io/jenkins/plugins/remotingkafka/builder/ProducerPropertiesBuilder.java

```diff
@@ -56,21 +57,21 @@ public ProducerPropertiesBuilder withValueSerialier(Class valueSerialier) {
         return this;
     }
 
-    public Properties build() {
+    public Properties build() throws RemotingKafkaConfigurationException {
         Properties props = new Properties();
         if (bootsrapServers == null) {
-            throw new IllegalStateException("Please provide Kafka producer bootstrap servers");
+            throw new RemotingKafkaConfigurationException("Please provide Kafka producer bootstrap servers");
```

Comment:
this logic could be refactored later, e.g. by `checkAndPut(Properties target, String varName, String displayName, Object var) throws RemotingKafkaConfigurationException` or so